src/orquestra/sdk/_base/_storage/orqdantic.py

```diff
@@ -0,0 +1,123 @@
+################################################################################
+# © Copyright 2024 Zapata Computing Inc.
+################################################################################
+
+"""Compatibility layer for pydantic v1 / v2 compatibility."""
+
+from copy import deepcopy
+from typing import Annotated, Any, Optional
+
+import pydantic
+
+PYDANTICV1 = pydantic.__version__.startswith("1.")
+
+if PYDANTICV1:
+    from pydantic.generics import GenericModel
+
+    class BaseModel(GenericModel):
+        @classmethod
+        def model_validate(cls, *args, **kwargs):
+            return super(BaseModel, cls).parse_obj(*args, **kwargs)
+
+        @classmethod
+        def model_validate_json(cls, *args, **kwargs):
+            return super(BaseModel, cls).parse_raw(*args, **kwargs)
+
+        def model_dump(self, *args, **kwargs):
+            return super().dict(*args, **kwargs)
+
+        def model_dump_json(self, *args, **kwargs):
+            return super().json(*args, **kwargs)
+
+        @classmethod
+        def model_json_schema(cls, *args, **kwargs):
+            return super(BaseModel, cls).schema_json(*args, **kwargs)
+
+        def model_copy(self, *args, **kwargs):
+            return super().copy(*args, **kwargs)
+
+else:
+    # TODO (ORQSDK-1025): remove the model base class and replace it with an alies to
+    # BaseModel
+    class BaseModel(pydantic.BaseModel):
+        """The pydantic BaseModel changed between V1 and V2.
+
+        As a result, workflow outputs generated prior to the V2 upgrade may not be
+        depickled correctly. The culpret is a change in behaviour of `__setstate__`.
+
+        This class adds a new `__setstate__` that wraps the V2 BaseModel `__setstate__`
+        and adds the missing behaviour back in.
+        """
+
+        def __setstate__(self, state: dict[Any, Any]) -> None:
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏Technically, we still support 3.8 so this should be:
```suggestion
        def __setstate__(self, state: Dict[Any, Any]) -> None:
```
And import `Dict` from `typing`.